common/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -417,26 +432,21 @@ public boolean isDocumentPending(@NonNull String docId) throws CouchbaseLiteExce
                 CBLError.Code.UNSUPPORTED);
         }
 
-        final C4Replicator repl;
-        synchronized (lock) { repl = c4Replicator; }
-
-        // ??? Not yet implemented
-        if (repl == null) {
-            Log.i(DOMAIN, "%s: Call to isDocumentPending with unstarted replicator", this);
-            return false;
-        }
-
         final Boolean pending;
-        try { pending = repl.isDocumentPending(docId); }
+        try { pending = lazyCreateC4Replicator().isDocumentPending(docId); }
         catch (LiteCoreException e) { throw CBLStatus.convertException(e, "Failed getting document pending status"); }
 
-        // ??? C4 call returns null if the native replicator is gone: Not yet implemented
-        return (pending != null) && pending;
+        if (pending == null) { throw new IllegalStateException("Pending doc ids is unexpectedly null"); }
```

Comment:
What is the case that this could happen? It doesn't throw LiteCoreException when it happens?

Reply:
Also the native one returns boolean not Boolean: 
https://github.com/couchbase/couchbase-lite-java-common/blob/64cd753e44fce6ef15bb3084cd940018c5c12032/common/main/java/com/couchbase/lite/internal/core/C4Replicator.java#L510

Reply:
Yes.  Nice catch.  This code is leftover from when the C4Replicator could be freed.  Fixing.